redis/src/sentinel.rs

```diff
@@ -117,15 +117,17 @@ use crate::aio::MultiplexedConnection as AsyncConnection;
 use crate::client::AsyncConnectionConfig;
 
 use crate::{
-    connection::ConnectionInfo, types::RedisResult, Client, Cmd, Connection, ErrorKind,
-    FromRedisValue, IntoConnectionInfo, RedisConnectionInfo, TlsMode, Value,
+    connection::{ActualConnection, ConnectionInfo},
+    types::RedisResult,
+    Client, Cmd, Connection, ConnectionDriver, ErrorKind, FromRedisValue, GenericConnection,
+    IntoConnectionInfo, RedisConnectionInfo, TlsMode, Value,
 };
 
 /// The Sentinel type, serves as a special purpose client which builds other clients on
 /// demand.
-pub struct Sentinel {
+pub struct Sentinel<IO = ActualConnection> {
```

Comment:
I'm not sure that this will work when adding sans-io support for aio work in sentinel - the users will need to declare 2 generic types, even if they care only about one, right?

Reply:
I'm not sure i follow here. Also I realize after contributing to this crate that my redis-client usage was pretty simple and minimal.

Can you give some more context around `Sentinel` and also explain why this might be an issue for async IO? Is it because sentinel handles both the async api and the regular one? Would it perhaps make sense to split that up, to no longer do that if that is the case?

Reply:
>  Is it because sentinel handles both the async api and the regular one?

Exactly. it contains both `async_connections_cache` and `connections_cache`. Adding the generic boundary to `connections_cache` implies that there will be a separate generic boundary for `async_connections_cache`, which will complicate their usage.
In any case, for backwards compatibility you'll have to use the same trick you used for `Connection` - Implement a `GenericSentinel<IO>` and then define `pub struct Sentinel(GenericSentinel<Connection>)`.

In any case, I think that you don't need to implement sentinel initially. I think It's a niche feature, and it can keep using `Connection` until there will be a request to implement it using sans-io, too.

Reply:
> Would it perhaps make sense to split that up, to no longer do that if that is the case?

Maybe, but again, I'd like to avoid backwards incompatibility.